src/blr/Snakefile

```diff
@@ -32,12 +32,29 @@ rule starcode_clustering:
         " 2> {log}"
 
 
+rule barcode_sort_fastq:
+    # Assumes barcode read name is followed by barcode sequence.
+    # Exmaple: @ST-E00269:339:H27G2CCX2:7:1102:21186:8060:AAAAAAAATATCTACGCTCA BX:Z:AAAAAAAATATCTACGCTCA
+    output:
+        fastq = "sorted.{nr}.fastq.gz"
+    input:
+        fastq = "trimmed.barcoded.{nr}.fastq.gz"
+    shell:
+        "pigz -cd {input.fastq} |"
+        " paste - - - - |"
+        " awk -F ' ' '{{print $2,$0}}' |"
+        " sort -t ' ' -k1,1  |"
+        " cut -d' ' -f 2- |"
+        " tr '\t' '\n' |"
+        " gzip > {output.fastq}"
```

Comment:
I think the `awk` can be removed and I think it makes sense to use `pigz` for zipping. 
```
"pigz -cd {input.fastq} |"
" paste - - - - |"
" sort -t "_" -k 3 |"
" tr "\t" "\n" |"
" pigz > {output.fastq}"
```

Results from a 100x blr-testdata-0.2 sorting.

```
time pigz -cd 100x-testfile.fq.gz | paste - - - - | sort -t "_" -k 3 | tr "\t" "\n" | pigz > sort.fq.gz

real	1m39.001s
user	1m46.161s
sys	0m2.255s
```
```
time pigz -cd 100x-testfile.fq.gz | paste  - - - | awk -F ' ' '{{print $2,$0}}' | sort -t ' ' -k1,1  | cut -d' ' -f 2- | tr '\t' '\n' | gzip > sort.fq.gz 

real	1m45.886s
user	2m16.831s
sys	0m2.180s
```

Reply:
When using `pigz` for decompression, please use it in single-thread mode (add option `-p 1`). According to the man page, decompression cannot be parallelized, so it just spawns some extra I/O threads which help to reduce wall-clock time. However, this comes at the cost of higher total CPU time. What actually *is* faster than `gzip` for some weird reason is `pigz -dc -p 1`, which seems to use a more efficient (albeit still single-threaded) decompression algorithm.

Reply:
Thanks for the comments, I will add changes! 👍